File: day_2/2.js
function createFrame(names) {
  const maxLength = Math.max(...names.map((name) => name.trim().length));
  const frameWidth = maxLength + 4;
  const frame = "*".repeat(frameWidth);
  const namesWithPadding = names.map(
    (name) => `* ${name.trim().padEnd(maxLength)} *`
  );
  return [frame, ...namesWithPadding, frame].join("\n");
}

console.log(createFrame(["midu", "madeval", "educalvolpz"]));
console.log(createFrame(["midu"]));
console.log(createFrame(["a", "bb", "ccc"]));
console.log(createFrame(["a", "bb", "ccc", "dddd"]));

File: day_2/2.py
def createFrame(names):
    max_length = 0
    for name in names:
        if len(name) > max_length:
            max_length = len(name)
    box = ""
    for name in names:
        box += "* " + name + " " * (max_length - len(name)) + "*\n"

    return "*" * (max_length + 4) + "\n" + box + "*" * (max_length + 4)


print(createFrame(["midu", "madeval", "educalvolpz"]))
